lib/metrics.js

```diff
@@ -21,7 +21,17 @@ function RedisMetrics(config) {
 
   this.config = config = config || {};
 
-  if (config.host && config.port) {
+  if (this.config.client) {
+    if (this.config.client.constructor.name !== 'RedisClient') {
```

Comment:
I think `instanceof` will actually be better here because it will ensure that the library is exactly the same?
```javascript
...
if (this.config.client instanceof redis.RedisClient) {
...
```